apps/examples/testcase/le_tc/kernel/tc_libc_sched.c

```diff
@@ -82,6 +84,54 @@ static void tc_libc_sched_sched_get_priority_min(void)
 	TC_SUCCESS_RESULT();
 }
 
+/**
+* @fn                   :function_wait
+* @description          :Function for tc_sched_wait
+* @return               :int
+*/
+static int function_wait(int argc, char *argv[])
+{
+	usleep(USECVAL);
+	task_delete(0);
+	return 0;
+}
+
+/**
+ * @fn                   :tc_libc_sched_task_setcanceltype
+ * @brief                :This tc tests tc_libc_sched_task_setcanceltype()
+ * @Scenario             :The task_setcanceltype() function atomically both sets the calling
+ *                        task's cancelability type to the indicated type and returns the
+ *                        previous cancelability type at the location referenced by oldtype
+ *                        If successful pthread_setcanceltype() function shall return zero;
+ *                        otherwise, an error number shall be returned to indicate the error.
+ * @API'scovered         :task_setcanceltype
+ * @Preconditions        :none
+ * @Postconditions       :none
+ * @return               :void
+ */
+#ifndef CONFIG_CANCELLATION_POINTS
+static void tc_libc_sched_task_setcanceltype(void)
+{
+	int type;
+	int oldtype;
+	int ret_chk;
+	pid_t pid;
+
+	/* creating new process */
+	pid = task_create("sched1", SCHED_PRIORITY_DEFAULT, TASK_STACKSIZE, function_wait, (char * const *)NULL);
+	TC_ASSERT_GT("task_create", pid, OK);
+
+	type = TASK_CANCEL_ASYNCHRONOUS;
```

Comment:
You should move all the below logic to new task execution context , ie function_wait()   and it's not advised to call task_delete(pid)  in the parent task context, as it's already taken care inside function_wait()